airbyte-webapp/src/hooks/services/BulkEdit/BulkEditService.test.tsx

```diff
@@ -0,0 +1,236 @@
+import { renderHook } from "@testing-library/react-hooks";
+import { Formik } from "formik";
+import React from "react";
+import { act } from "react-dom/test-utils";
+
+import { SyncSchemaStream } from "../../../core/domain/catalog";
+import { DestinationSyncMode, SyncMode } from "../../../core/request/AirbyteClient";
+import { TestWrapper } from "../../../test-utils/testutils";
+import { BulkEditServiceProvider, useBulkEditSelect, useBulkEditService } from "./BulkEditService";
+
+const MOCK_NODES = [
+  {
+    id: "1",
+    stream: {
+      sourceDefinedCursor: true,
+      defaultCursorField: ["source_cursor"],
+      sourceDefinedPrimaryKey: [["new_primary_key"]],
+      jsonSchema: {},
+      name: "test",
+      namespace: "namespace-test",
+      supportedSyncModes: [],
+    },
+    config: {
+      destinationSyncMode: DestinationSyncMode.append,
+      selected: false,
+      syncMode: SyncMode.full_refresh,
+      cursorField: ["old_cursor"],
+      primaryKey: [["old_primary_key"]],
+      aliasName: "",
+    },
+  },
+  {
+    id: "2",
+    stream: {
+      sourceDefinedCursor: true,
+      defaultCursorField: ["source_cursor"],
+      sourceDefinedPrimaryKey: [["new_primary_key"]],
+      jsonSchema: {},
+      name: "test2",
+      namespace: "namespace-test-2",
+      supportedSyncModes: [],
+    },
+    config: {
+      destinationSyncMode: DestinationSyncMode.append,
+      selected: false,
+      syncMode: SyncMode.full_refresh,
+      cursorField: ["old_cursor"],
+      primaryKey: [["old_primary_key"]],
+      aliasName: "",
+    },
+  },
+];
+const mockUpdate = jest.fn();
+
+const provider = (
+  nodes: SyncSchemaStream[],
+  update: (streams: SyncSchemaStream[]) => void
+): React.FC<React.PropsWithChildren<unknown>> => {
+  return ({ children }) => (
+    <TestWrapper>
+      <Formik initialValues={{}} onSubmit={async () => false} initialStatus={{ editControlsVisible: false }}>
+        <BulkEditServiceProvider nodes={nodes} update={update}>
+          {children}
+        </BulkEditServiceProvider>
+      </Formik>
+    </TestWrapper>
+  );
+};
+
+describe("BulkEditServiceProvider", () => {
+  beforeEach(() => {
+    mockUpdate.mockClear();
+  });
+  it("isActive should work correctly", () => {
```

Comment:
"work correctly" always changes and the tests should describe what they do. Example for this test:

"isActive should be true when toggleNode is changed"